### PR TITLE
2021 02 23 commitalert work

### PR DESCRIPTION
**Pull request by tomrittervg**
the base branch of this PR is another branch so that the diff is easier to review (you won't see the commits in the branch.) But when we finally merge it it should be changed to merge into master.

### CODE REVIEW

File: automation.py [hunk 1]
```diff
@@ -143,15 +143,38 @@ def runOnProviders(self, func):
             func(v)
 
     def _validate(self, config_dictionary):
+        # In this function we have not set up our robust logging facilities yet
```
Comment (metalcanine) [inline automation.py:146]: I'm not sure what this means, is it that we haven't moved from our basic logger to the full loggingprovider yet?

File: automation.py [hunk 2]
```diff
@@ -143,15 +143,38 @@ def runOnProviders(self, func):
...
+            try:
+                with open(os.path.join(config_dictionary['General']['gecko-path'], "browser", "config", "version.txt")) as version_file:
+                    version = version_file.read()
+                    ff_version = int(version.split(".")[0])
+            except Exception as e:
+                self.logger.log("Encountered an error trying to read the version from version.txt", level=LogLevel.Fatal)
+                raise e
+
+            if ff_version < 87:
+                self.logger.log("The FF version we pulled from the repo is < 87: %s" % ff_version, level=LogLevel.Fatal)
```
Comment (metalcanine) [inline automation.py:172]: Can we add a comment on why < 87 is gated from running updatebot?

File: components/libraryprovider.py [hunk 3]
```diff
@@ -101,19 +101,23 @@ def get_key_or_default(key, dict, default):
...
-        mozilla_central_yamls = self.run(["find", gecko_path, "-type", "f", "-name", "moz.yaml"]).stdout.decode().strip().split("\n")
-
-        for file in mozilla_central_yamls:
-            with open(file, "r") as mozyaml:
-                # Only return libraries that have enabled tasks
-                new_library_obj = LibraryProvider.validate_library(mozyaml.read(), file.replace(gecko_path + "/", ""))
-                if new_library_obj.tasks:
-                    libraries.append(new_library_obj)
-        return libraries
+        if self._libraries is None:
```
Comment (metalcanine) [inline components/libraryprovider.py:107]: Oh neat! I didn't realize that caching the result was going to be that small of a change. Nice!

**Review by metalcanine — CHANGES_REQUESTED** [review]
A question about a comment and a needed comment in one spot to help ourselves out in the future, but otherwise looks great!

**Comment by tomrittervg** [comment]
I've completly changed this branch to be commits that are relatively concrete and isolated from the work of creating a commit alert task.

**Review by metalcanine — APPROVED** [review]
Looks great, thanks!

**Pull request merged**